contracts/token/ERC721/base/ERC721Base.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {AddressUtils} from '../../../utils/AddressUtils.sol';
+import {EnumerableMap} from '../../../utils/EnumerableMap.sol';
+import {EnumerableSet} from '../../../utils/EnumerableSet.sol';
+import {IERC721} from '../IERC721.sol';
+import {IERC721Receiver} from '../IERC721Receiver.sol';
+import {ERC721BaseStorage} from './ERC721BaseStorage.sol';
+import {ERC721BaseInternal} from './ERC721BaseInternal.sol';
+
+/**
+ * @notice Base ERC721 implementation, excluding optional extensions
+ */
+abstract contract ERC721Base is IERC721, ERC721BaseInternal {
+  using AddressUtils for address;
+  using EnumerableMap for EnumerableMap.UintToAddressMap;
+  using EnumerableSet for EnumerableSet.UintSet;
+
+  /**
+   * @inheritdoc IERC721
+   */
+  function balanceOf (address account) override public view returns (uint) {
+    require(account != address(0), 'ERC721: balance query for the zero address');
+    return ERC721BaseStorage.layout().holderTokens[account].length();
+  }
+
```

Comment:
Split with internal `_balanceOf` function, for consistency with ERC1155 ?

Reply:
I think yes, but where do we stop?  Should every external function have a corresponding internal function?